uniffi_meta/src/lib.rs

```diff
@@ -309,11 +309,33 @@ pub struct FieldMetadata {
     pub docstring: Option<String>,
 }
 
+#[repr(u8)]
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Checksum)]
+pub enum EnumFlatness {
+    // It's an error and variants are exposed.
+    ErrorWithVariants,
+    // It's an error but variants are not exposed to the error enum
+    ErrorFlat,
+    // It's a regular enum, so flatness is based on the actual variants.
+    EnumValues,
+}
```

Comment:
This is way nicer than before, but I'd love to do some more bikeshedding on the names/variants.  How do you feel about:
  -  `EnumStyle` instead of `EnumFlatness`.  I think this is now capturing more than just the flatness.
  - `Enum` instead of `EnumValues`.  To me, "Values" indicates that there are associated values, but I think this would be the variant for a non-error enum with no associated values.
  - `Error { is_flat: bool }` instead of the 2 error variants?  I'm not totally sure about this suggestion, but I think I like it.

Reply:
I think I quite like that suggestion in general, but I'll need to ponder more, because:

> I think this is now capturing more than just the flatness.

currently is really is just about the flatness.

Reply:
I was thinking about [this block](https://github.com/mozilla/uniffi-rs/pull/2119/files#diff-554b47625fdbd9d813c73ea0e8feb461e1bf447d5496682b6cefb30c33b92d80R818) where we can also use the value to check if it's used as an error or not.

Reply:
Yeah, I was thinking a little more about this and you are 100% correct

Reply:
That worked out really well, thanks. I bike-shedded even more and ended up with `EnumShape` but I think it looks good!